sfall/AnimationsAtOnceLimit.cpp

```diff
@@ -143,160 +143,181 @@ static const DWORD sad_28[] = {
 	0x4173CE, 0x4174C1, 0x4175F1, 0x417730,
 };
 
-void AnimationsAtOnceInit(signed char aniMax) {
-
-	if (aniMax <= 32) return;
-
-	AniLimitFixActive = true;
-
-	int i;
-
-	//allocate memory to store larger animation struct arrays
-	anim_set = new BYTE[2656*(aniMax+1)];
-	sad = new BYTE[3240*(aniMax+1)];
-
-	//set general animation limit check (old 20) aniMax-12 -- +12 reserved for PC movement(4) + other critical animations(8)?
-	SafeWrite8(0x413C07, aniMax-12);
-
-	//PC movement animation limit checks (old 24) aniMax-8 -- +8 reserved for other critical animations?.
-	for (i = 0; i < sizeof(AnimPCMove)/4; i++) {
-		SafeWrite8(AnimPCMove[i], aniMax-8);
-	}
-
-	//Max animation limit checks (old 32) aniMax
-	for (i = 0; i < sizeof(AnimMaxCheck)/4; i++) {
-		SafeWrite8(AnimMaxCheck[i], aniMax);
-	}
-
-	//Max animations checks - animation struct size * max num of animations (old 2656*32=84992)
-	for (i = 0; i < sizeof(AnimMaxSizeCheck)/4; i++) {
-		SafeWrite32(AnimMaxSizeCheck[i], 2656*aniMax);
-	}
-
-
-	//divert old animation structure list pointers to newly alocated memory
-
-	//struct array 1///////////////////
-
-	//old addr 0x54C1B4
-	SafeWrite32(0x413A9E, (DWORD)anim_set);
-
-	//old addr 0x54C1C0
-	for (i = 0; i < sizeof(fake_anim_set_C)/4; i++) {
-		SafeWrite32(fake_anim_set_C[i], 12+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC14
-	for (i = 0; i < sizeof(anim_set_0)/4; i++) {
-		SafeWrite32(anim_set_0[i], 2656+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC18
-	for (i = 0; i < sizeof(anim_set_4)/4; i++) {
-		SafeWrite32(anim_set_4[i], 2656+4+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC1C
-	for (i = 0; i < sizeof(anim_set_8)/4; i++) {
-		SafeWrite32(anim_set_8[i], 2656+8+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC20
-	for (i = 0; i < sizeof(anim_set_C)/4; i++) {
-		SafeWrite32(anim_set_C[i], 2656+12+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC24
-	for (i = 0; i < sizeof(anim_set_10)/4; i++) {
-		SafeWrite32(anim_set_10[i], 2656+16+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC28
-	for (i = 0; i < sizeof(anim_set_14)/4; i++) {
-		SafeWrite32(anim_set_14[i], 2656+20+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC38
-	SafeWrite32(0x413F29, 2656+36+(DWORD)anim_set);
-
-	//old addr 0x54CC3C
-	for (i = 0; i < sizeof(anim_set_28)/4; i++) {
-		SafeWrite32(anim_set_28[i], 2656+40+(DWORD)anim_set);
-	}
-
-	//old addr 0x54CC48
-	SafeWrite32(0x415C35, 2656+52+(DWORD)anim_set);
-
-
-	//struct array 2///////////////////
-
-	//old addr 0x530014
-	for (i = 0; i < sizeof(sad_0)/4; i++) {
-		SafeWrite32(sad_0[i], (DWORD)sad);
-	}
-
-	//old addr 0x530018
-	for (i = 0; i < sizeof(sad_4)/4; i++) {
-		SafeWrite32(sad_4[i], 4+(DWORD)sad);
-	}
-
-	//old addr 0x53001C
-	for (i = 0; i < sizeof(sad_8)/4; i++) {
-		SafeWrite32(sad_8[i], 8+(DWORD)sad);
+static void __declspec(naked) anim_set_end_hook() {
+	__asm {
+		mov  edi, _anim_set
+		cmp  dword ptr AnimationsLimit, 32
+		jbe  skip
```

Comment:
This should be "jle" instead of jbe; otherwise the game will crash upon entering combat mode if anyone is stupid enough to set AnimationsAtOnceLimit to negative values.